### DISCUSSION TITLE
Feature of fighting back players (#211 is not solved)

### DISCUSSION THREAD

**Discussion by Ch3225** [seymourimadeit/guardvillagers#218] [Ideas]
**further reply in #211 might be ignored by you, please check.**
You introduced FriendlyFire mod, it works but it will also introduce other problems, like making guards fully invincible. You didn't replied me on my further reply in #211 , so I just tried fixing this problem on my own by adding some codes.
**That's why I've modified the code to implement this and made a pull request #216.**  please also check it, which could be a solution of this issue. **

**Comment by seymourimadeit** [comment]
Sorry for not checking the replies in #211, life is busy and I had to focus on other things.
As for #216 I do like the idea in theory but I think it could just lead to abuse as it stands currently (insta-killing guards for armor while they don't attack you because you technically don't have a low reputation to get free armor). 
Of course, you could make that feature an addon to the mod using the same methods just inside a forge event instead.

I do have a plan for dealing with this though in the future where it's based off a 3 strike type of thing but I'm open to suggestions

**Reply by Ch3225** [reply]
How about applying it but adding another mechanic: Guard would only drop their equipment only if they are died from a mob. 
The logic is like the following:
```
when(guard died){
     if(guard is hurted by a player a few seconds ago    or     guard is killed by a player){
         guard won't drop equipment;
     }else{
         if(guard died because of a mob){
             guard drop equipment;
         }else{
             guard won't drop equipment;
         }
     }
}
```
So it would be no more easy for players to get free armors, but I am wondering if this logic could be easily implemented technically. 
In some emergency situations (such as raid events or there's more difficult monsters defined by other modules attacking the village), players can also obtain equipment dropped by guards. Although there is a possibility that players deliberately let the guards die to mobs and then pick up their equipment, this is still very time-consuming and hard-to-control, hence it could be no more a big problem. If not, well, we can let guards spawn only with equipment up to iron ones.
